### PR TITLE
Feature/metadata and metadata filtering

### PR DESCRIPTION
**Pull request by allenan**
This should be good to go.

### CODE REVIEW

File: lib/helium/collection.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      self
+    end
+
+    # Returns an array of the Resources belonging to the Collection
+    # @return [Resource]
+    def collection
+      fetch_collection
+    end
+
+    def each
```
Comment (allenan) [inline lib/helium/collection.rb]: each is the minimum method that needs to be defined to implement the `Enumerable` mixin. Kind of like an interface in obj-c, but just not enforced.

Comment (allenan) [inline lib/helium/collection.rb]: good overview that goes more in depth than what we're doing: https://blog.codeship.com/the-enumerable-module/

Comment (jaredmorrow) [inline lib/helium/collection.rb]: I figured as much b/c the rest of enumerable just worked.  I just couldn't find an interface definition like you'd see in java/c++ that states the bare minimum you need to overload the operators.

**Comment by allenan** [comment]
i'm going to go ahead and merge & release this and regenerate the docs.

I already see at least one place in dashboard where we'll need to make a small adjustment, so we'll need to make sure that we test it pretty thoroughly after upgrading the gem version

**Pull request merged**